### ISSUE TITLE
REXML::ParseException when Partial Has More Than 1 Root

### ISSUE THREAD

**Issue by keokilee** [zurb/inky-rb#11]
When I include a partial that includes more than one top level element, I get a `REXML::ParseException`.

My partial is:

```
<div>Partial content 1</div>
<div>Partial content 2</div>
```

And here's a partial stack trace:

```
#<RuntimeError: attempted adding second root element to document>
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/document.rb:95:in `add'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/element.rb:879:in `add'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/child.rb:21:in `initialize'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/parent.rb:13:in `initialize'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/element.rb:59:in `initialize'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/element.rb:877:in `new'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/element.rb:877:in `add'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/element.rb:297:in `add_element'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/document.rb:102:in `add_element'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/parsers/treeparser.rb:33:in `parse'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/document.rb:287:in `build'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/2.2.0/rexml/document.rb:44:in `initialize'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/inky-rb-1.3.6.1/lib/inky.rb:35:in `new'
/Users/keokilee/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/inky-rb-1.3.6.1/lib/inky.rb:35:in `release_the_kraken'
/Users/keokilee/Code/InkyTest/app/views/inky_mailer/_inky_partial.html.inky:1:in `_app_views_inky_mailer__inky_partial_html_inky__4142895012036450841_70155048147260'
```

As a workaround, we're just wrapping the content of the partial with an outer `div`.

**Comment by kball** [comment]
I think we need to make inky-rb smart enough to insert a root node when it needs one.  I believe this can be handled by catching exactly this exception and wrapping the content in a div... we can probably even re-remove it after the end.

I'll hopefully have a chance to tackle this shortly, but would happily accept a PR that implements this. 

If you want to tackle it, I'd add a test for it in the components_spec and then look at making this modification in the `release_the_kraken` method of lib/inky.rb.

**Comment by marcandre** [comment]
I think this has been fixed in 1.3.6.2, now that we use `nokogiri` instead of `REXML`. @keokilee Is it easy for you to re-test this?

**Comment by marcandre** [comment]
Actually, it's not fixed yet, sorry...

**Comment by kball** [comment]
I'm checking with our HTML emails expert about the XHTML/HTML question for emails, but in the meantime I'd thought one possible workaround for this is to wrap everything in a root element prior to parsing that we then strip out after transformation.

**Comment by marcandre** [comment]
@kball Check #19d8f83. Using an HTML parser instead of XML fixes a whole bunch of issues, including this one.

**Comment by marcandre** [comment]
Should be completely addressed (for real! :smile:) with [1.3.7.0](https://github.com/zurb/inky-rb/tree/v1.3.7.0). Let us know if that's not the case! Thanks for reporting this.
